From 9c556145a2814a858479ccf223f0fbaa8fbe5b1e Mon Sep 17 00:00:00 2001
From: Vinayakumar B <vinayakumarb@apache.org>
Date: Mon, 22 Feb 2016 09:21:46 +0530
Subject: [PATCH 1330/2848] HDFS-7452. skip StandbyException log for
 getCorruptFiles() (Contributed by Bramha Reddy
 Battula)

(cherry picked from commit f313516731d787f6be64c9406ca83d941d47ee99)
(cherry picked from commit 3e76768f216772dbb7e1ec2e9055eed57f45fee9)
(cherry picked from commit 8ca8d218a480246813f1029ceec361fce0f5ef02)

Change-Id: I9fd55f0dc6522cd65cb3419a272b8b9014cc0db9
---
 .../hadoop/hdfs/server/namenode/FSNamesystem.java  |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSNamesystem.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSNamesystem.java
index 84bdd66..fa7f966 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSNamesystem.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSNamesystem.java
@@ -8032,6 +8032,10 @@ public String getCorruptFiles() {
           list.add(c.toString());
         }
       }
+    } catch (StandbyException e) {
+      if (LOG.isDebugEnabled()) {
+        LOG.debug("Get corrupt file blocks returned error: " + e.getMessage());
+      }
     } catch (IOException e) {
       LOG.warn("Get corrupt file blocks returned error: " + e.getMessage());
     }
-- 
1.7.9.5

